kv/etcd/store.go

```diff
@@ -389,10 +389,11 @@ func (c *client) getFromKVStore(key string) (kv.Value, error) {
 func (c *client) getFromEtcdEvents(key string, events []*clientv3.Event) kv.Value {
 	lastEvent := events[len(events)-1]
 	if lastEvent.Type == clientv3.EventTypeDelete {
+		c.deleteCache(key)
```

Comment:
This was the fix? cc @cw9

Reply:
Aye. We handled cache eviction correctly for deletes in the client doing the delete. But did not handle it correctly for anyone watching on the key. Without this change, on receiving an update we early terminated without updating the cache.

Reply:
I vaguely recall that we used to specifically not delete cache on watch updates, to prevent mistakenly deleting a key (e.g placement). This will change that assumption, I'm not sure if there are other use cases in m3db that relies on that assumption.

Reply:
Also to for the case Matt mentioned, you might need to call `deleteCache` on line 406 as well

Reply:
re: not deleting - after the stale cache issue Matt found in prod, i think we should add this change for safety. if m3db/other clients needs not apply deletes they should do so in the application layer (e.g. what we do for namespace removes - we see the update and choose not to act on it).

Reply:
re: extra deleteCache call, cheers - that helps narrow my spelunking :)

Reply:
@cw9 interestingly found something else that i don't understand. `initCache` decodes the local cache file, but if it succeeds at that, it doesn't load the cache with those values. What was the intent there? 

see test: `TestStaleDelete` to see what I mean.